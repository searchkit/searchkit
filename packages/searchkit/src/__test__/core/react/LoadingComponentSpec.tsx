import * as React from "react";

import {
  SearchkitManager,
  LoadingComponent
} from "../../../"

import {mount} from "enzyme";

describe("SearchkitProvider", ()=> {

  beforeEach(()=> {
    this.searchkit = SearchkitManager.mock()
    this.wrapper = mount(
      <LoadingComponent searchkit={this.searchkit}>
        <p>loading...</p>
      </LoadingComponent>
    )
  })

  it("display nothing when not loading", ()=> {
    expect(this.wrapper).toMatchSnapshot()
  })

  it("display children when loading", ()=> {
    this.searchkit.loading = true
    this.wrapper.instance().forceUpdate()
    expect(this.wrapper).toMatchSnapshot()
  })


})
