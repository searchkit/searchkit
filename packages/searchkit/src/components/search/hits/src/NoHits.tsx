import * as PropTypes from "prop-types";

import {
	SearchkitComponent,
	SearchkitComponentProps,
	NoFiltersHitCountAccessor,
	SuggestionsAccessor,
	RenderComponentType,
	renderComponent
} from "../../../../core"

import { NoHitsErrorDisplay, NoHitsErrorDisplayProps } from "./NoHitsErrorDisplay"
import { NoHitsDisplay, NoHitsDisplayProps } from "./NoHitsDisplay"

const defaults = require("lodash/defaults")

export interface NoHitsProps extends SearchkitComponentProps {
	suggestionsField?: string
	errorComponent?: RenderComponentType<NoHitsErrorDisplayProps>
	component?: RenderComponentType<NoHitsDisplayProps>
}

export class NoHits extends SearchkitComponent<NoHitsProps, any> {
	noFiltersAccessor: NoFiltersHitCountAccessor
	suggestionsAccessor: SuggestionsAccessor
	bemBlocks: {
		container: Function
	}

	static translations = {
		"NoHits.NoResultsFound": "No results found for {query}.",
		"NoHits.NoResultsFoundDidYouMean": "No results found for {query}. Did you mean {suggestion}?",
		"NoHits.DidYouMean": "Search for {suggestion} instead",
		"NoHits.SearchWithoutFilters": "Search for {query} without filters",
		"NoHits.Error": "We're sorry, an issue occurred when fetching your results. Please try again.",
		"NoHits.ResetSearch": "Reset Search"
	}
	translations = NoHits.translations

	static propTypes = defaults({
		suggestionsField: PropTypes.string,
		translations: SearchkitComponent.translationsPropType(
			NoHits.translations
		)
	}, SearchkitComponent.propTypes)

	static defaultProps = {
		errorComponent: NoHitsErrorDisplay,
		component: NoHitsDisplay
	}

	componentDidMount() {
		super.componentDidMount()
		this.noFiltersAccessor = this.searchkit.addAccessor(
			new NoFiltersHitCountAccessor()
		)
		if (this.props.suggestionsField) {
			this.suggestionsAccessor = this.searchkit.addAccessor(
				new SuggestionsAccessor(this.props.suggestionsField)
			)
		}
	}

	defineBEMBlocks() {
		let block = (this.props.mod || "sk-no-hits")
		return {
			container: block
		}
	}

	getSuggestion() {
		return this.suggestionsAccessor && this.suggestionsAccessor.getSuggestion()
	}

	setQueryString(query) {
		this.searchkit.getQueryAccessor().setQueryString(query, true)
		this.searchkit.performSearch(true)
	}

	resetFilters() {
		this.searchkit.getQueryAccessor().keepOnlyQueryState()
		this.searchkit.performSearch(true)
	}

	resetSearch() {
		this.searchkit.getQueryAccessor().resetState()
		this.searchkit.performSearch(true)
	}

	getFilterCount() {
		return this.noFiltersAccessor && this.noFiltersAccessor.getCount()
	}

	render() : React.ReactElement<any> {
		if ((this.hasHits() || this.isInitialLoading() || this.isLoading()) && !this.getError()) return null

		if (this.getError()) {
			const props: NoHitsErrorDisplayProps = {
				errorLabel: this.translate("NoHits.Error"),
				resetSearchFn: this.resetSearch.bind(this),
				translate: this.translate,
				bemBlocks: this.bemBlocks,
				tryAgainLabel: this.translate("NoHits.ResetSearch"),
				error: this.getError()
			}
			return renderComponent(this.props.errorComponent, props)
		}

		const suggestion: any = this.getSuggestion()
		const query = this.getQuery().getQueryString()
		let infoKey = suggestion ? "NoHits.NoResultsFoundDidYouMean" : "NoHits.NoResultsFound"

		const props: NoHitsDisplayProps = {
			noResultsLabel: this.translate(infoKey, { query: query, suggestion: suggestion }),
			translate: this.translate,
			bemBlocks: this.bemBlocks,
			suggestion: suggestion,
			query: query,
			filtersCount: this.getFilterCount(),
			resetFiltersFn: this.resetFilters.bind(this),
			setSuggestionFn: this.setQueryString.bind(this, suggestion)
		}

		return renderComponent(this.props.component, props)

	}
}
