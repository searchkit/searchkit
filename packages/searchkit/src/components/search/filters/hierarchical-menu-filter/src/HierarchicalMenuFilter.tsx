import * as React from "react";
import * as PropTypes from "prop-types";

import {
	SearchkitComponent,
	HierarchicalFacetAccessor,
	FastClick,
	SearchkitComponentProps,
	renderComponent,
	RenderComponentType,
	RenderComponentPropType
} from "../../../../../core"

import {
	Panel, ItemComponent, ItemProps
} from "../../../../ui"

const defaults = require("lodash/defaults")
const map = require("lodash/map")
const identity = require("lodash/identity")

export interface HierarchicalMenuFilterProps extends SearchkitComponentProps {
	id: string
	fields: Array<string>
	title: string
	size?: number
	orderKey?: string
	orderDirection?: string
	countFormatter?: (count: number) => string | number
	containerComponent?: RenderComponentType<any>
	itemComponent?: RenderComponentType<ItemProps>
}

export class HierarchicalMenuFilter extends SearchkitComponent<HierarchicalMenuFilterProps, any> {
	public accessor: HierarchicalFacetAccessor

	static defaultProps = {
		countFormatter: identity,
		size: 20,
		containerComponent:Panel,
		itemComponent:ItemComponent
	}
	static propTypes = defaults({
		id: PropTypes.string.isRequired,
		fields: PropTypes.arrayOf(PropTypes.string).isRequired,
		title: PropTypes.string.isRequired,
		orderKey: PropTypes.string,
		orderDirection: PropTypes.oneOf(["asc", "desc"]),
		countFormatter: PropTypes.func,
		containerComponent: RenderComponentPropType,
		itemComponent: RenderComponentPropType
	}, SearchkitComponent.propTypes)

	defineBEMBlocks() {
		var blockClass = this.props.mod || "sk-hierarchical-menu";
		return {
			container: `${blockClass}-list`,
			option: `${blockClass}-option`
		};
	}

	defineAccessor() {
		const { id, title, fields, size, orderKey, orderDirection } = this.props
		return new HierarchicalFacetAccessor(id, {
			id, title, fields, size, orderKey, orderDirection
		})
	}

	addFilter(option, level) {
		this.accessor.state = this.accessor.state.toggleLevel(
			level, option.key)

		this.searchkit.performSearch()
	}

	renderOption(level, option) {

		var block = this.bemBlocks.option
		const { countFormatter, itemComponent } = this.props
		const active = this.accessor.state.contains(level, option.key) 
		
		return (
			<div key={option.key}>
				{renderComponent(itemComponent, {
					active,
					itemKey:option.key,
					showCount:true,
					bemBlocks:this.bemBlocks,
					onClick:this.addFilter.bind(this, option, level),
					label:this.translate(option.key),
					count:countFormatter(option.doc_count)
				})}
				{(() => {
					if (this.accessor.resultsState.contains(level, option.key)) {
						return this.renderOptions(level + 1);
					}
				})()}
			</div>
		)
	}

	renderOptions(level) {
		let block = this.bemBlocks.container;
		return (
			<div className={block("hierarchical-options")}>
				{map(this.accessor.getBuckets(level), this.renderOption.bind(this, level))}
			</div>
		)
	}

	render() {
		if(!this.accessor) return null;
		const block = this.bemBlocks.container
		const { id, title, containerComponent } = this.props
		return renderComponent(
			containerComponent, {
				title,
				className: id ? `filter--${id}` : undefined,
				disabled: this.accessor.getBuckets(0).length == 0
			}, 
				<div className={block("root")}>
					{this.renderOptions(0)}
				</div>
			
		)
	}
}
