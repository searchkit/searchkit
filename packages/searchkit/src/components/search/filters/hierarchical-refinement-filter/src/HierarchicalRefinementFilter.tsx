import * as React from "react"
import * as PropTypes from "prop-types"

import {
	SearchkitComponent,
	NestedFacetAccessor,
	FastClick,
	SearchkitComponentProps,
	RenderComponentType,
	renderComponent,
	RenderComponentPropType
} from "../../../../../core"

import {
	Panel, ItemComponent, ItemProps
} from "../../../../ui"

const defaults = require("lodash/defaults")
const map = require("lodash/map")
const identity = require("lodash/identity")

export interface HierarchicalRefinementFilterProps extends SearchkitComponentProps {
	field: string
	id: string
	title: string
	size?: number
	orderKey?: string
	orderDirection?: string
	startLevel?: number
	countFormatter?: (count: number) => number | string
	containerComponent?: RenderComponentType<any>
	itemComponent?: RenderComponentType<ItemProps>
}

export class HierarchicalRefinementFilter extends SearchkitComponent<HierarchicalRefinementFilterProps, any> {
	public accessor: NestedFacetAccessor

	static defaultProps = {
		countFormatter: identity,
		containerComponent: Panel,
		itemComponent: ItemComponent
	}

	static propTypes = defaults({
		field: PropTypes.string.isRequired,
		id: PropTypes.string.isRequired,
		title: PropTypes.string.isRequired,
		orderKey: PropTypes.string,
		orderDirection: PropTypes.oneOf(["asc", "desc"]),
		startLevel: PropTypes.number,
		countFormatter: PropTypes.func,
		containerComponent: RenderComponentPropType,
		itemComponent: RenderComponentPropType
	}, SearchkitComponent.propTypes)

	defineBEMBlocks() {
		var blockClass = this.props.mod || "sk-hierarchical-refinement";
		return {
			container: `${blockClass}-list`,
			option: `${blockClass}-option`
		}
	}

	defineAccessor() {
		const {
			field, id, title, size, orderKey,
			orderDirection, startLevel 
		} = this.props;

		return new NestedFacetAccessor(id, {
			field, id, title, size, orderKey,
			orderDirection, startLevel
		})
	}

	addFilter(level, option) {
		this.accessor.state = this.accessor.state.toggleLevel(level, option.key)
		this.searchkit.performSearch()
	}

	renderOption(level, option) {

		var block = this.bemBlocks.option
		var active = this.accessor.resultsState.contains(level, option.key)
		const { countFormatter, itemComponent } = this.props		

		return (
			<div key={option.key}>
				{renderComponent(itemComponent, {
					active,
					bemBlocks:this.bemBlocks,
					label:this.translate(option.key),
					itemKey:option.key,
					count:countFormatter(option.doc_count),
					showCount:true,
					onClick: this.addFilter.bind(this, level, option)
				})}
				{active && this.renderOptions(level + 1)}					
			</div>
		)
	}

	renderOptions(level) {
		let block = this.bemBlocks.container;
		return (
			<div className={block("hierarchical-options")}>
				{
					map(this.accessor.getBuckets(level), this.renderOption.bind(this, level))
				}
			</div>
		)
	}

	render() {
		if(!this.accessor) return null;
		const block = this.bemBlocks.container
		const { id, title, containerComponent } = this.props
		return renderComponent(
			containerComponent, {
				title,
				className: id ? `filter--${id}` : undefined,
				disabled: this.accessor.getBuckets(0).length == 0
			},
			<div className={block("root")}>
				{this.renderOptions(0)}
			</div>

		)
	}


}
