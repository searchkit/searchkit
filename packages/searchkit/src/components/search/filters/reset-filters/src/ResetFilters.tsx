import * as React from "react";
import * as PropTypes from "prop-types";

import {
	SearchkitComponent,
	FastClick,
	SearchkitComponentProps,
	RenderComponentType,
	ResetSearchOptions,
	ResetSearchAccessor,
  renderComponent
} from "../../../../../core"
const defaults = require("lodash/defaults")


export interface ResetFiltersDisplayProps {
	bemBlock:any,
	hasFilters:boolean,
	resetFilters:Function,
	clearAllLabel:string,
	translate:Function
}

export class ResetFiltersDisplay extends React.PureComponent<ResetFiltersDisplayProps, any>{
	render(){
		const { bemBlock, hasFilters, resetFilters, clearAllLabel } = this.props
		return (
			<div>
				<FastClick handler={resetFilters}>
					<div className={bemBlock().state({disabled:!hasFilters})}>
						<div className={bemBlock("reset")}>{clearAllLabel}</div>
					</div>
				</FastClick>
			</div>
		)
	}
}

export interface ResetFiltersProps extends SearchkitComponentProps {
	component?: RenderComponentType<ResetFiltersDisplayProps>,
	options?:ResetSearchOptions
}

export class ResetFilters extends SearchkitComponent<ResetFiltersProps, any> {

	static translations:any = {
		"reset.clear_all":"Clear all filters"
	}
	translations = ResetFilters.translations
	accessor:ResetSearchAccessor

	static propTypes = defaults({
		translations:SearchkitComponent.translationsPropType(
			ResetFilters.translations
		),
		component:PropTypes.func,
		options:PropTypes.object
	}, SearchkitComponent.propTypes)

	static defaultProps = {
		component:ResetFiltersDisplay
	}

	constructor(props){
		super(props)
		this.resetFilters = this.resetFilters.bind(this)
	}

	defineBEMBlocks() {
		return {
			container: (this.props.mod || "sk-reset-filters")
		}
	}

	defineAccessor(){
		return new ResetSearchAccessor(this.props.options)
	}

	resetFilters() {
		this.accessor.performReset()
		this.searchkit.performSearch()
	}

  render() {
	if(!this.accessor) return null;
		const props:ResetFiltersDisplayProps = {
			bemBlock:this.bemBlocks.container,
			resetFilters:this.resetFilters,
			hasFilters:this.accessor.canReset(),
			translate:this.translate,
			clearAllLabel: this.translate("reset.clear_all")
		}
		return renderComponent(this.props.component, props)
  }
}
