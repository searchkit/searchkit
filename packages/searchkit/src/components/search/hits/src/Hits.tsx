import * as React from "react";
import * as PropTypes from "prop-types";

import {
	SearchkitComponent,
	PageSizeAccessor,
	HighlightAccessor,
	CustomHighlightAccessor,
	SearchkitComponentProps,
	SourceFilterType,
	SourceFilterAccessor,
	HitsAccessor,
	RenderComponentType,
	RenderComponentPropType,
	renderComponent,
	block
} from "../../../../core"

const map = require("lodash/map")
const defaults = require("lodash/defaults")



export interface HitItemProps {
	key:string,
	bemBlocks?:any,
	result:any
}

export class HitItem extends React.PureComponent<HitItemProps, any> {

	render(){
		return (
			<div data-qa="hit"
				className={this.props.bemBlocks.item().mix(this.props.bemBlocks.container("item"))}>
				{this.props.result._id}
			</div>
		)
	}
}

export interface HitsListProps{
	mod?:string,
	className?:string,
	itemComponent?:RenderComponentType<HitItemProps>,
	hits:Array<Object>
}

export class HitsList extends React.PureComponent<HitsListProps, any>{

	static defaultProps={
		mod:"sk-hits",
		itemComponent:HitItem
	}

	static propTypes = {
		mod:PropTypes.string,
		className:PropTypes.string,
		itemComponent:RenderComponentPropType,
		hits:PropTypes.any
	}

	render(){
		const {hits, mod, className, itemComponent} = this.props
		const bemBlocks = {
			container: block(mod).el,
			item: block(`${mod}-hit`).el
		}
		return (
			<div data-qa="hits" className={bemBlocks.container().mix(className)}>
				{map(hits, (result: any, index)=> {
					return renderComponent(itemComponent, {
						key:result._id, result, bemBlocks, index
					})
				})}
			</div>
		)
	}
}

export interface HitsProps extends SearchkitComponentProps{
	hitsPerPage?: number
	highlightFields?:Array<string>
	customHighlight?:any
	sourceFilter?:SourceFilterType
	itemComponent?: RenderComponentType<HitItemProps>
	listComponent?: RenderComponentType<HitsListProps>
	scrollTo?: boolean|string
}


export class Hits extends SearchkitComponent<HitsProps, any> {
	hitsAccessor:HitsAccessor

	static propTypes = defaults({
		hitsPerPage:PropTypes.number,
		highlightFields:PropTypes.arrayOf(
			PropTypes.string
		),
		sourceFilterType:PropTypes.oneOf([
			PropTypes.string,
			PropTypes.arrayOf(PropTypes.string),
			PropTypes.bool
		]),
		itemComponent:RenderComponentPropType,
		listComponent:RenderComponentPropType
	}, SearchkitComponent.propTypes)

	static defaultProps = {
		listComponent:HitsList,
		scrollTo: "body"
	}

	componentDidMount() {
		super.componentDidMount()
		if(this.props.hitsPerPage){
			this.searchkit.getAccessorByType(PageSizeAccessor)
				.defaultSize = this.props.hitsPerPage
		}
		if(this.props.highlightFields) {
			this.searchkit.addAccessor(
				new HighlightAccessor(this.props.highlightFields))
		}
		if (this.props.customHighlight) {
			this.searchkit.addAccessor(new CustomHighlightAccessor(this.props.customHighlight))
		}
		if(this.props.sourceFilter){
			this.searchkit.addAccessor(
				new SourceFilterAccessor(this.props.sourceFilter)
			)
		}
		this.hitsAccessor = new HitsAccessor({ scrollTo:this.props.scrollTo })
		this.searchkit.addAccessor(this.hitsAccessor)
	}


	render() {
		let hits:Array<Object> = this.getHits()
		let hasHits = hits.length > 0

		if (!this.isInitialLoading() && hasHits) {
			const {listComponent, mod, className, itemComponent} = this.props
			return renderComponent(listComponent, {
				hits, mod, className, itemComponent
			})
		}

		return null

	}
}
