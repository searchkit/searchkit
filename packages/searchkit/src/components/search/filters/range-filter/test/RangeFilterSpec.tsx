import * as React from "react";
import {mount} from "enzyme";
import {RangeFilter} from "../src/RangeFilter";
import { RangeSliderHistogram, RangeInput } from "../../../../../components"
import {SearchkitManager, RangeAccessor} from "../../../../../core";
import {
  fastClick
} from "../../../../__test__/TestHelpers"


import * as sinon from "sinon";

describe("Range Filter tests", () => {

  beforeEach(() => {

    this.searchkit = SearchkitManager.mock()
    spyOn(this.searchkit, "performSearch")
    this.rangeFormatter = (count) => count + " score"
    this.createWrapper = ({
      withHistogram=true, 
      interval=undefined, 
      rangeComponent=RangeSliderHistogram
    }) => {

      
      this.wrapper = mount(
        <RangeFilter
          id="m"
          searchkit={this.searchkit}
          field="metascore"
          min={0}
          max={100}
          title="metascore"
          interval={interval}
          translations={{"range.divider":" to "}}
          rangeFormatter={this.rangeFormatter}
          rangeComponent={rangeComponent}
          showHistogram={withHistogram}/>
      );


      this.searchkit.setResults({
        "aggregations": {
          "m1": {
            "m": {
              "buckets": [
                { key: "10", doc_count: 1 },
                { key: "20", doc_count: 3 },
                { key: "30", doc_count: 1 },
                { key: "40", doc_count: 1 },
                { key: "50", doc_count: 1 },
                { key: "60", doc_count: 5 },
                { key: "70", doc_count: 1 },
                { key: "80", doc_count: 1 },
                { key: "90", doc_count: 1 },
                { key: "100", doc_count: 1 }
              ],
              "value":10
            }
          }
        }
      })


     

      this.wrapper.update()
      this.accessor = this.searchkit.getAccessorByType(RangeAccessor)
    }


  });


  it("accessor has correct config", () => {
    this.createWrapper({withHistogram:true})
    expect(this.accessor.options).toEqual({
      id:"m",
      min:0,
      max:100,
      field:"metascore",
      title:"metascore",
      interval: undefined,
      loadHistogram:true,
      fieldOptions:{
        type:'embedded',
        field:'metascore'
      },
      rangeFormatter:this.rangeFormatter,
      translations:{"range.divider": " to "}
    })
  })

  it('renders correctly', () => {
    this.createWrapper({withHistogram:true})    
    expect(this.wrapper).toMatchSnapshot()
  })

  it("renders without histogram", () => {
    this.createWrapper({withHistogram:false})
    expect(this.wrapper.find(".sk-range-histogram").length).toBe(0)
    expect(this.wrapper.find(".sk-range-histogram__bar").length).toBe(0)
    expect(this.wrapper).toMatchSnapshot()
  })

  it("handle slider events correctly", ()=> {
    this.createWrapper(true)
    this.wrapper.instance().sliderUpdate({min:30,max:70})
    expect(this.accessor.state.getValue()).toEqual({
      min:30, max:70
    })
    expect(this.searchkit.performSearch).not.toHaveBeenCalled()

    this.wrapper.instance().sliderUpdateAndSearch({min:40,max:60})
    expect(this.accessor.state.getValue()).toEqual({
      min:40, max:60
    })
    expect(this.searchkit.performSearch).toHaveBeenCalled()
    let query = this.searchkit.buildQuery()
    expect(query.getSelectedFilters()[0].value).toEqual('40 score to 60 score')
    expect(this.wrapper).toMatchSnapshot()

    // min/max should clear
    this.wrapper.instance().sliderUpdateAndSearch({min:0,max:100})
    expect(this.accessor.state.getValue()).toEqual({})
    expect(this.wrapper).toMatchSnapshot()

  })

  it("has default interval", ()=> {
    this.createWrapper({withHistogram:true})
    expect(this.accessor.getInterval()).toEqual(5)
  })

  it("handles interval correctly", ()=> {
    this.createWrapper({ withHistogram: true, interval:2 })
    expect(this.accessor.getInterval()).toEqual(2)
  })

  it("renders limited range correctly", ()=> {
    this.createWrapper({ withHistogram: true })

    this.wrapper.instance().sliderUpdate({min:30,max:70})
    expect(this.wrapper).toMatchSnapshot()
  })

  it("renders with range input component", ()=> {
    this.createWrapper({ withHistogram: false, rangeComponent:RangeInput })
    expect(this.wrapper).toMatchSnapshot()       
    
    this.wrapper.find("input[placeholder='min']")
      .simulate('change', {target:{value:20}})    
    this.wrapper.find("input[placeholder='max']")
      .simulate('change', { target: { value: 80 } })    
    this.wrapper.find("form").simulate('submit')
    let query = this.searchkit.buildQuery()
    expect(query.getSelectedFilters()[0].value)
      .toEqual("20 score to 80 score")
    
    

  })

});
