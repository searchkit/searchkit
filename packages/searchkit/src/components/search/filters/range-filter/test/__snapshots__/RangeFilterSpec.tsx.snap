// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Range Filter tests handle slider events correctly 1`] = `
<div class="sk-panel filter--m">
  <div class="sk-panel__header">metascore</div>
  <div class="sk-panel__content">
    <div>
      <div class="sk-range-histogram">
        <div class="sk-range-histogram__bar is-out-of-bounds" style="height: 20%;"></div>
        <div class="sk-range-histogram__bar is-out-of-bounds" style="height: 60%;"></div>
        <div class="sk-range-histogram__bar is-out-of-bounds" style="height: 20%;"></div>
        <div class="sk-range-histogram__bar" style="height: 20%;"></div>
        <div class="sk-range-histogram__bar" style="height: 20%;"></div>
        <div class="sk-range-histogram__bar" style="height: 100%;"></div>
        <div class="sk-range-histogram__bar is-out-of-bounds" style="height: 20%;"></div>
        <div class="sk-range-histogram__bar is-out-of-bounds" style="height: 20%;"></div>
        <div class="sk-range-histogram__bar is-out-of-bounds" style="height: 20%;"></div>
        <div class="sk-range-histogram__bar is-out-of-bounds" style="height: 20%;"></div>
      </div>
      <div class="sk-range-slider">
        <div class="rc-slider rc-slider-with-marks">
          <div class="rc-slider-rail"></div>
          <div class="rc-slider-track rc-slider-track-1" style="left: 40%; width: 20%;"></div>
          <div class="rc-slider-step"><span class="rc-slider-dot" style="left: 0%;"></span><span class="rc-slider-dot" style="left: 100%;"></span></div>
          <div role="slider" tabindex="0" aria-valuemin="0" aria-valuemax="100" aria-valuenow="40" aria-disabled="false" class="rc-slider-handle rc-slider-handle-1"
            style="left: 40%;"></div>
          <div role="slider" tabindex="0" aria-valuemin="0" aria-valuemax="100" aria-valuenow="60" aria-disabled="false" class="rc-slider-handle rc-slider-handle-2" style="left: 60%;"></div>
          <div class="rc-slider-mark"><span class="rc-slider-mark-text" style="width: 90%; margin-left: -45%; left: 0%;">0 score</span><span class="rc-slider-mark-text" style="width: 90%; margin-left: -45%; left: 100%;">100 score</span></div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Range Filter tests handle slider events correctly 2`] = `
<div class="sk-panel filter--m">
  <div class="sk-panel__header">metascore</div>
  <div class="sk-panel__content">
    <div>
      <div class="sk-range-histogram">
        <div class="sk-range-histogram__bar" style="height: 20%;"></div>
        <div class="sk-range-histogram__bar" style="height: 60%;"></div>
        <div class="sk-range-histogram__bar" style="height: 20%;"></div>
        <div class="sk-range-histogram__bar" style="height: 20%;"></div>
        <div class="sk-range-histogram__bar" style="height: 20%;"></div>
        <div class="sk-range-histogram__bar" style="height: 100%;"></div>
        <div class="sk-range-histogram__bar" style="height: 20%;"></div>
        <div class="sk-range-histogram__bar" style="height: 20%;"></div>
        <div class="sk-range-histogram__bar" style="height: 20%;"></div>
        <div class="sk-range-histogram__bar" style="height: 20%;"></div>
      </div>
      <div class="sk-range-slider">
        <div class="rc-slider rc-slider-with-marks">
          <div class="rc-slider-rail"></div>
          <div class="rc-slider-track rc-slider-track-1" style="left: 0%; width: 100%;"></div>
          <div class="rc-slider-step"><span class="rc-slider-dot rc-slider-dot-active" style="left: 0%;"></span><span class="rc-slider-dot rc-slider-dot-active" style="left: 100%;"></span></div>
          <div role="slider" tabindex="0" aria-valuemin="0" aria-valuemax="100" aria-valuenow="0"
            aria-disabled="false" class="rc-slider-handle rc-slider-handle-1" style="left: 0%;"></div>
          <div role="slider" tabindex="0" aria-valuemin="0" aria-valuemax="100" aria-valuenow="100" aria-disabled="false" class="rc-slider-handle rc-slider-handle-2" style="left: 100%;"></div>
          <div class="rc-slider-mark"><span class="rc-slider-mark-text rc-slider-mark-text-active" style="width: 90%; margin-left: -45%; left: 0%;">0 score</span><span class="rc-slider-mark-text rc-slider-mark-text-active" style="width: 90%; margin-left: -45%; left: 100%;">100 score</span></div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Range Filter tests renders correctly 1`] = `
<div class="sk-panel filter--m">
  <div class="sk-panel__header">metascore</div>
  <div class="sk-panel__content">
    <div>
      <div class="sk-range-histogram">
        <div class="sk-range-histogram__bar" style="height: 20%;"></div>
        <div class="sk-range-histogram__bar" style="height: 60%;"></div>
        <div class="sk-range-histogram__bar" style="height: 20%;"></div>
        <div class="sk-range-histogram__bar" style="height: 20%;"></div>
        <div class="sk-range-histogram__bar" style="height: 20%;"></div>
        <div class="sk-range-histogram__bar" style="height: 100%;"></div>
        <div class="sk-range-histogram__bar" style="height: 20%;"></div>
        <div class="sk-range-histogram__bar" style="height: 20%;"></div>
        <div class="sk-range-histogram__bar" style="height: 20%;"></div>
        <div class="sk-range-histogram__bar" style="height: 20%;"></div>
      </div>
      <div class="sk-range-slider">
        <div class="rc-slider rc-slider-with-marks">
          <div class="rc-slider-rail"></div>
          <div class="rc-slider-track rc-slider-track-1" style="left: 0%; width: 100%;"></div>
          <div class="rc-slider-step"><span class="rc-slider-dot rc-slider-dot-active" style="left: 0%;"></span><span class="rc-slider-dot rc-slider-dot-active" style="left: 100%;"></span></div>
          <div role="slider" tabindex="0" aria-valuemin="0" aria-valuemax="100" aria-valuenow="0"
            aria-disabled="false" class="rc-slider-handle rc-slider-handle-1" style="left: 0%;"></div>
          <div role="slider" tabindex="0" aria-valuemin="0" aria-valuemax="100" aria-valuenow="100" aria-disabled="false" class="rc-slider-handle rc-slider-handle-2" style="left: 100%;"></div>
          <div class="rc-slider-mark"><span class="rc-slider-mark-text rc-slider-mark-text-active" style="width: 90%; margin-left: -45%; left: 0%;">0 score</span><span class="rc-slider-mark-text rc-slider-mark-text-active" style="width: 90%; margin-left: -45%; left: 100%;">100 score</span></div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Range Filter tests renders limited range correctly 1`] = `
<div class="sk-panel filter--m">
  <div class="sk-panel__header">metascore</div>
  <div class="sk-panel__content">
    <div>
      <div class="sk-range-histogram">
        <div class="sk-range-histogram__bar is-out-of-bounds" style="height: 20%;"></div>
        <div class="sk-range-histogram__bar is-out-of-bounds" style="height: 60%;"></div>
        <div class="sk-range-histogram__bar" style="height: 20%;"></div>
        <div class="sk-range-histogram__bar" style="height: 20%;"></div>
        <div class="sk-range-histogram__bar" style="height: 20%;"></div>
        <div class="sk-range-histogram__bar" style="height: 100%;"></div>
        <div class="sk-range-histogram__bar" style="height: 20%;"></div>
        <div class="sk-range-histogram__bar is-out-of-bounds" style="height: 20%;"></div>
        <div class="sk-range-histogram__bar is-out-of-bounds" style="height: 20%;"></div>
        <div class="sk-range-histogram__bar is-out-of-bounds" style="height: 20%;"></div>
      </div>
      <div class="sk-range-slider">
        <div class="rc-slider rc-slider-with-marks">
          <div class="rc-slider-rail"></div>
          <div class="rc-slider-track rc-slider-track-1" style="left: 30%; width: 40%;"></div>
          <div class="rc-slider-step"><span class="rc-slider-dot" style="left: 0%;"></span><span class="rc-slider-dot" style="left: 100%;"></span></div>
          <div role="slider" tabindex="0" aria-valuemin="0" aria-valuemax="100" aria-valuenow="30" aria-disabled="false" class="rc-slider-handle rc-slider-handle-1"
            style="left: 30%;"></div>
          <div role="slider" tabindex="0" aria-valuemin="0" aria-valuemax="100" aria-valuenow="70" aria-disabled="false" class="rc-slider-handle rc-slider-handle-2" style="left: 70%;"></div>
          <div class="rc-slider-mark"><span class="rc-slider-mark-text" style="width: 90%; margin-left: -45%; left: 0%;">0 score</span><span class="rc-slider-mark-text" style="width: 90%; margin-left: -45%; left: 100%;">100 score</span></div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Range Filter tests renders with range input component 1`] = `
<div class="sk-panel filter--m">
  <div class="sk-panel__header">metascore</div>
  <div class="sk-panel__content">
    <form class="sk-range-input"><input type="number" class="sk-range-input__input" placeholder="min" value="0">
      <div class="sk-range-input__to-label">-</div><input type="number" class="sk-range-input__input" placeholder="max" value="100"><button type="submit" class="sk-range-input__submit">Go</button></form>
  </div>
</div>
`;

exports[`Range Filter tests renders without histogram 1`] = `
<div class="sk-panel filter--m">
  <div class="sk-panel__header">metascore</div>
  <div class="sk-panel__content">
    <div class="sk-range-slider">
      <div class="rc-slider rc-slider-with-marks">
        <div class="rc-slider-rail"></div>
        <div class="rc-slider-track rc-slider-track-1" style="left: 0%; width: 100%;"></div>
        <div class="rc-slider-step"><span class="rc-slider-dot rc-slider-dot-active" style="left: 0%;"></span><span class="rc-slider-dot rc-slider-dot-active" style="left: 100%;"></span></div>
        <div role="slider" tabindex="0" aria-valuemin="0" aria-valuemax="100" aria-valuenow="0"
          aria-disabled="false" class="rc-slider-handle rc-slider-handle-1" style="left: 0%;"></div>
        <div role="slider" tabindex="0" aria-valuemin="0" aria-valuemax="100" aria-valuenow="100" aria-disabled="false" class="rc-slider-handle rc-slider-handle-2" style="left: 100%;"></div>
        <div class="rc-slider-mark"><span class="rc-slider-mark-text rc-slider-mark-text-active" style="width: 90%; margin-left: -45%; left: 0%;">0 score</span><span class="rc-slider-mark-text rc-slider-mark-text-active" style="width: 90%; margin-left: -45%; left: 100%;">100 score</span></div>
      </div>
    </div>
  </div>
</div>
`;
